src/pipeline.cpp

```diff
@@ -545,6 +545,7 @@ static void buildIntrinsicLoweringPipeline(ModulePassManager &MPM, PassBuilder *
             JULIA_PASS(FPM.addPass(LateLowerGCPass()));
             JULIA_PASS(FPM.addPass(FinalLowerGCPass()));
             if (O.getSpeedupLevel() >= 2) {
+                FPM.addPass(DSEPass());
```

Comment:
Should these pipeline changes go in a separate PR?

Reply:
maybe. It is somewhat closely related since this PR introduces the possibility for a bunch of extra dead stores to be revealed.